### PR TITLE
PMM-5272 Add --socket flag for add mysql commands.

### PR DESCRIPTION
**Pull request by BupycHuk**
https://github.com/Percona-Lab/pmm-submodules/pull/773

### CODE REVIEW

File: commands/inventory/add_service_mysql.go [hunk 1]
```diff
@@ -84,6 +93,22 @@ func (cmd *addServiceMySQLCommand) Run() (commands.Result, error) {
 	}, nil
 }
 
+func (cmd *addServiceMySQLCommand) validateParams() error {
+	if cmd.Socket != "" {
+		if cmd.Address != "" || cmd.Port != 0 {
+			return fmt.Errorf("both socket and address are passed")
+		}
+	} else {
```
Comment (askomorokhov) [inline commands/inventory/add_service_mysql.go]: `else` seems rudimental

File: commands/management/add_mysql.go [hunk 2]
```diff
@@ -115,6 +117,44 @@ func (cmd *addMySQLCommand) GetAddress() string {
...
+	if *addServiceNameFlag != "" {
+		serviceName = *addServiceNameFlag
+	}
+
+	socket = cmd.Socket
+	address := cmd.GetAddress()
+	var portI int
+	if socket != "" && (address != "" || *addHostFlag != "" || *addPortFlag != 0) {
+		return "", "", "", 0, fmt.Errorf("both socket and address are passed")
+	} else if socket == "" {
```
Comment (askomorokhov) [inline commands/management/add_mysql.go:131]: I guess here `else` we can remove as well.

File: commands/management/add_mysql.go [hunk 3]
```diff
@@ -115,6 +117,44 @@ func (cmd *addMySQLCommand) GetAddress() string {
...
+		if address == "" {
+			address = "127.0.0.1:3306"
+		}
+		var portS string
+		host, portS, err = net.SplitHostPort(address)
```
Comment (askomorokhov) [inline commands/management/add_mysql.go:136]: We need to add a check if the port is in the valid range, otherwise, we hide the error.

**Review by askomorokhov — CHANGES_REQUESTED** [review]
Can we cover this with tests?

File: commands/inventory/add_service_mysql.go (hunk 1 above)
Comment (BupycHuk) [reply commands/inventory/add_service_mysql.go]: no, it's not, you can see that not all branches above are return result

File: commands/management/add_mysql.go (hunk 2 above)
Comment (BupycHuk) [reply commands/management/add_mysql.go:131]: do you think it's better just leave `if socket == ""`?

File: commands/management/add_mysql.go (hunk 3 above)
Comment (BupycHuk) [reply commands/management/add_mysql.go:136]: We don't check it anywhere :) Good point

**Review by percona-csalguero — APPROVED** [review]

**Review by artemgavrilov — APPROVED** [review]

File: commands/inventory/add_service_mysql.go [hunk 4]
```diff
@@ -93,8 +119,9 @@ var (
 func init() {
 	AddServiceMySQLC.Arg("name", "Service name").StringVar(&AddServiceMySQL.ServiceName)
 	AddServiceMySQLC.Arg("node-id", "Node ID").StringVar(&AddServiceMySQL.NodeID)
-	AddServiceMySQLC.Arg("address", "Address").StringVar(&AddServiceMySQL.Address)
-	AddServiceMySQLC.Arg("port", "Port").Int64Var(&AddServiceMySQL.Port)
+	AddServiceMySQLC.Arg("address", "Address").Default("").StringVar(&AddServiceMySQL.Address)
+	AddServiceMySQLC.Arg("port", "Port").Default("").Int64Var(&AddServiceMySQL.Port)
```
Comment (askomorokhov) [inline commands/inventory/add_service_mysql.go]: Why we need an empty string as the default value for a port?

Comment (BupycHuk) [reply commands/inventory/add_service_mysql.go]: To make it optional argument and differentiate when user set the value "3306" manually

**Review by askomorokhov — APPROVED** [review]

**Pull request merged**